packages/amplify-category-geo/src/service-utils/validateGeoJSONFile.ts

```diff
@@ -0,0 +1,83 @@
+import { readFileSync } from 'fs-extra';
+import Ajv from 'ajv';
+import { FeatureCollection, IdentifierOption } from './populateParams';
+import GeoJSONSchema from 'amplify-category-geo/schema/GeoJSONSchema.json';
+import { v4 as uuid } from 'uuid';
+import { printer } from 'amplify-prompts';
+
+
+export const validateGeoJSONFile = (geoJSONFilePath: string, uniqueIdentifier: string = 'id', identifierOption: IdentifierOption = IdentifierOption.RootLevelID) => {
+  const data = JSON.parse(readFileSync(geoJSONFilePath, 'utf-8')) as FeatureCollection;
+  //Validate against pre-defined schema
+  const ajv = new Ajv();
+  const validator = ajv.compile(GeoJSONSchema);
+  if (!validator(data) as boolean) {
+    throw new Error(`Data did not validate against the supplied schema. Underlying errors were ${JSON.stringify(validator.errors)}`);
+  };
+  const { features } = data;
+  let identifierSet = new Set();
+  features.forEach((feature) => {
+    //Check for identifier uniqueness
+    let identifierField: string;
+    if (identifierOption === IdentifierOption.RootLevelID) {
+      if (!feature.id) {
+        feature.id = uuid();
+        printer.info(`No root level id found. Auto assigned feature with id ${feature.id}.`)
+      }
+      identifierField = feature.id
+    }
+    else {
+      identifierField = feature.properties[uniqueIdentifier];
+      //Check if custom identifier exists in property
+      if (!identifierField) {
+        throw new Error(`Identifier field ${uniqueIdentifier} is missing in the feature property`)
+      }
+    }
+    if (identifierSet.has(identifierField)) {
+      throw new Error(`Identifier field ${uniqueIdentifier} is not unique in GeoJSON.`)
+    }
+    identifierSet.add(identifierField);
+    //Additional validation for each linear ring
+    const { coordinates } = feature.geometry;
+    coordinates.forEach((linearRing, index) => {
+      validateLinearRing(linearRing, index === 0);
+    })
+  });
+  return data;
+}
+
+/**
+ * Validate linear ring for those not captured by JSON Schema, incluidng first and last identical position, winding direction
+ * @param linearRing Array of coordinates
+ * @param isFirstRing Whether it is first(exterior) ring
+ */
+const validateLinearRing = (linearRing: Array<Array<number>>, isFirstRing: boolean) => {
+  const numPoint = linearRing.length;
+  //Check if first position is identical to last one
+  if (!(linearRing[0][0] === linearRing[numPoint-1][0] && linearRing[0][1] === linearRing[numPoint-1][1])) {
+    throw new Error(`Linear ring "${linearRing}" should have the identical values for first and last position.`);
```

Comment:
can we pass in the Geofence ID and throw a more useful error like `Linear ring "${linearRing}" should have the identical values for first and last position for the Geofence with ID ${geofenceID}`